test/jdk/com/sun/management/HotSpotDiagnosticMXBean/CheckOrigin.java

```diff
@@ -54,6 +54,7 @@ public static void main(String... args) throws Exception {
             // start a process that has options set in a number of different ways
 
             File flagsFile = File.createTempFile("CheckOriginFlags", null);
+            flagsFile.deleteOnExit();
```

Comment:
Hello Bill, jtreg uses a scratch directory when running tests. When a test is launched, the current working directory points to the scratch directory for the test that's currently executing. jtreg manages the lifecycle of scratch directories and even cleans them up (as necessary).
Would it instead be better to just create the temporary file within the jtreg scratch directory (represented by the current working directory)? That way you could just do:

```
File flagsFile = Files.createTempFile(Path.of("."), "CheckOriginFlags", null).toFile();
```
and don't need any explicit deletions?

Reply:
Hi Jaikiran, I think both solutions work for this bug. I personally prefer to place the files in the scratch directory for the ease of debugging. 
In addition, for this specific test, I am considering using File.createTempFile("CheckOriginaFlags", null, Path.of(".").toFile) instead of Files.createTempFile for consistency purposes, as Files.createTempFile may have more restrictive access permissions.